### PR TITLE
Replace TRI_voc_rid_t with RevisionId type

### PR DESCRIPTION
**Pull request by danielhlarkin**
### Scope & Purpose

Replace `TRI_voc_rid_t` with compiler-checkable `RevisionId` type.

- [x] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)

### Testing & Verification

This change is a trivial rework / code cleanup without any test coverage.

Jenkins: https://jenkins.arangodb.biz/job/arangodb-matrix-pr/9437/

### CODE REVIEW

File: arangod/Aql/SharedQueryState.cpp [hunk 1]
```diff
@@ -96,41 +96,41 @@ void SharedQueryState::queueHandler() {
     // We are shutting down
     return;
   }
-
```
Comment (graetzer) [inline arangod/Aql/SharedQueryState.cpp:99]: do you  have to commit this reformat ?

**Comment by danielhlarkin** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/11180/

**Comment by danielhlarkin** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/11204/

**Comment by danielhlarkin** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/11219/

File: arangod/VocBase/Identifiers/RevisionId.cpp [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+void RevisionId::toPersistent(std::string& buffer) const {
+  rocksutils::uint64ToPersistent(buffer, id());
+}
+
+/// @brief create a revision id using an HLC value
+RevisionId RevisionId::create() { return RevisionId{TRI_HybridLogicalClock()}; }
+
+/// @brief create a revision id which is guaranteed to be unique cluster-wide
+RevisionId RevisionId::createClusterWideUnique(application_features::ApplicationServer& server) {
+  ClusterFeature& cf = server.getFeature<ClusterFeature>();
```
Comment (graetzer) [inline arangod/VocBase/Identifiers/RevisionId.cpp:100]: do we really need that lookup everyhwere ? Can the code not use the ClusterInfo directly

**Comment by danielhlarkin** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/11234/

**Comment by danielhlarkin** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/11246/

**Comment by jsteemann** [comment]
There are some test failures that look as if they are caused by the changes in this PR:
https://172.16.10.101/job/arangodb-matrix-pr-linux/12922/EDITION=community,STORAGE_ENGINE=rocksdb,TEST_SUITE=single,limit=(linux&&test)%7C%7Cgce/artifact/testfailures.txt

**Comment by danielhlarkin** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/11271/

**Comment by jsteemann** [comment]
Tests blue

**Pull request merged**